iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NodeBreakerVoltageLevel.java

```diff
@@ -872,13 +863,23 @@ private void checkTerminal(TerminalExt terminal) {
         }
     }
 
+    private void addNodes(int node) {
+        int oldCount = graph.getVertexCount();
+        if (node >= oldCount) {
+            for (int i = oldCount; i < node + 1; i++) {
+                graph.addVertex();
+            }
+        }
+    }
```

Comment:
I'm not sure it's what we discussed. Adding a vertex will create a terminal. The idea is to create a terminal only if you need to attach something. As you remove the call to `clean()`, the unnecessary terminals are not removed.

It's compliant with the current version of powsybl, but I thought we wanted to go further.

Reply:
Okay, I was not really sure about the conclusion of what had been decided last time... If we want to create a vertex only each time a Terminal is created, we might not need a `clean` method (the vertex will be deleted when not used and created again if reused).